test/test.js

```diff
@@ -1,35 +1,457 @@
+    return JSON.parse(stringTestResult); //Return TestResult as JSON.
+}
+
+/**
+ * Returns a TestCycle object from a TestResult object by int.
+ * @param {JSON} testResult The JSON object containing the TestResults.
+ * @param {number} cycleInt The number defining the TestCycle.
+ * @returns {JSON} The JSON object containing the TestCycle.
+ */
+function getTestCycleByInt(testResult, cycleInt) {
+    //console.log(testResult[cycleInt])
+    return testResult[cycleInt];
+}
+
+/**
+ * Returns a TestScenario object from a TestCycle object by int.
+ * @param {JSON} testCycle The JSON object containing the TestCycle.
+ * @param {number} scenarioInt The number defining the TestScenario.
+ * @returns {JSON} The JSON object containing the TestScenario.
+ */
+function getTestScenarioByInt(testCycle, scenarioInt) {
+    return testCycle[scenarioInt];
+}
+
+/**
+ * Returns a TestCase object from a TestScenario object by int.
+ * @param {JSON} testScenario The JSON object containing the TestScenario.
+ * @param {JSON} testCaseInt The JSON object containing the TestCase.
+ * @return {JSON} The JSON object containing the TestCase.
+ */
+function getTestCaseByInt(testScenario, testCaseInt) {
+    return testScenario[testCaseInt];
+}
+
+/**
+ * Returns an array with all Test Cases in the given TestResult JSON object.
+ * @param {JSON} jsonTestResult The TestResult JSON.
+ * @return {[JSON]} The array containing all testCase JSON objects.
+ */
+function getAllTestCases(jsonTestResult) {
+    var allTestCases = [];
+    jsonTestResult.forEach((testCycle) => {
+        testCycle.forEach((testScenario) => {
+            //Could potentially be solved more elegant by using allTestCases.concat(testScenario), but
+            //then legacy loops would have to be used.
+            testScenario.forEach((testCase) => {
+                allTestCases.push(testCase);
+            });
+        });
+    });
+    return allTestCases;
+}
+
+/**
+ * Returns a TestCase JSON object from a TestScenario by name.
+ * @param {JSON} testScenario The JSON object containing the TestScenario.
+ * @param {string} testCaseName The string containing the TestCase name.
+ * @returns {JSON | null} The JSON object containing the TestCase or null if name was not found.
+ */
+function getTestCaseByName(testScenario, testCaseName) {
+    for (var i = 0; i < testScenario.length; ++i) {
+        if (testScenario[i] == testCaseName) {
+            return testScenario[i];
+        }
+        return null;
+    }
+}
+
+/**
+ * Returns all TestCase JSON objects with the given name from the given Array of TestCase JSON object.
+ * @param {[JSON]} testCases The array of JSON objects containing the TestCases to check.
+ * @param {string} testCaseName The string containing the TestCase name.
+ * @returns An arry containing all the TestCase JSON objects with the given name. Array is empty
+ * if no TestCase with the given name has been found.
+ */
+function getAllTestCasesWithName(testCases, testCaseName) {
```

Comment:
might be better to call byName to be similar to the function above

Reply:
I thought getAllTestCasesWithName makes more sence than getAllTestCasesByName. Would Also further differentiate the methods that return a single testCase from the ones that return multiple if in the future more methods like this are introduced. But can change it if you want.